src/getKeysets.ts

```diff
@@ -0,0 +1,109 @@
+import { Address, Chain, Hex, PublicClient, Transport, getAbiItem } from 'viem';
+import { sequencerInboxABI } from './abi';
+import { createRollupFetchTransactionHash } from './createRollupFetchTransactionHash';
+
+const SetValidKeysetEventAbi = getAbiItem({ abi: sequencerInboxABI, name: 'SetValidKeyset' });
+const InvalidateKeysetEventAbi = getAbiItem({ abi: sequencerInboxABI, name: 'InvalidateKeyset' });
+
+export type GetKeysetsParams = {
+  /** Address of the rollup we're getting list of keysets from */
+  rollup: Address;
+  /** Address of the sequencerInbox we're getting logs from */
+  sequencerInbox: Address;
```

Comment:
We don't need to pass the sequencerInbox as parameter, since we can obtain it from the Rollup address.

You can either call the `sequencerInbox` method in the rollupContract, or use the `createRollupFetchCoreContracts` method of the SDK.